src/modules/map/components/CompareSlider.vue

```diff
@@ -1,35 +1,71 @@
-<template>
-    <div class="compare-slider">
-        <font-awesome-icon
-            class="compare-slider-caret-left"
-            :icon="['fas', 'caret-up']"
-            :rotation="270"
-        />
-        <div class="compare-slider-line"></div>
-
-        <font-awesome-icon
-            class="compare-slider-caret-right"
-            :icon="['fas', 'caret-up']"
-            :rotation="90"
-        />
-    </div>
-</template>
-
-<script>
+<script setup>
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { computed, defineProps, onMounted, ref, toRefs, watch } from 'vue'
+import { useStore } from 'vuex'
 
-export default {
-    components: {
-        FontAwesomeIcon,
+const props = defineProps({
+    clientWidth: {
+        type: Number,
+        default: window.innerWidth,
     },
+})
+
+const { clientWidth } = toRefs(props)
+
+const compareRatio = ref(-0.5)
+
+const store = useStore()
+const storeCompareRatio = computed(() => store.state.ui.compareRatio)
+const compareSliderPosition = computed(() => compareRatio.value * 100 + '%')
+
+watch(storeCompareRatio, (newValue) => {
+    compareRatio.value = newValue
+})
+
+onMounted(() => {
+    compareRatio.value = storeCompareRatio.value
+})
+function generateCSSPosition() {
+    return { left: compareSliderPosition.value }
+}
+function grabSlider() {
+    window.addEventListener('mousemove', listenToMouseMove, { passive: true })
+    window.addEventListener('touchmove', listenToMouseMove, { passive: true })
+    window.addEventListener('mouseup', releaseSlider, { passive: true })
+    window.addEventListener('touchend', releaseSlider, { passive: true })
+}
+
+function listenToMouseMove(event) {
+    const currentPosition = event.type === 'touchmove' ? event.touches[0].pageX : event.pageX
+    compareRatio.value = currentPosition / clientWidth.value
```

Comment:
would be better to `round` to 2 decimal here (from `numberUtils.js`), the value in the URL can be very long otherwise